streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -345,6 +345,26 @@ boolean isProcessable(final long now) {
      */
     @SuppressWarnings("unchecked")
     public boolean process() {
+        // if condition put here in case of restarts and rebalances to check for correct timestamp
+        if (recordInfo.queue() != null && partitionGroup.getPartitionTimestamp(recordInfo.partition()) == -1) {
```

Comment:
nit: use RecordQueue.UNKNOWN instead of -1

Reply:
No problem, could fix that.